### PR TITLE
Use Configuration over MemoryManager for ImageFrame<TPixel>

### PR DESCRIPTION
**Pull request by JimBobSquarePants**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description
Fixes an issue we had where `CloneAs()` had to use the default configuration as it had no awareness of the current configuration. Simplifies the constructors a little now also.

### CODE REVIEW

**Comment by JimBobSquarePants** [comment]
WTF Travis!

**Comment by antonfirsov** [comment]
@JimBobSquarePants reopened #580 to test both CI builds with an empty PR.

**Comment by JimBobSquarePants** [comment]
There's a bug somewhere in the PR. 🙁 Tests are failing locally.

File: src/ImageSharp/ImageFrame{TPixel}.cs [hunk 1]
```diff
@@ -77,31 +84,31 @@ internal ImageFrame(Configuration configuration, int width, int height, TPixel b
...
 
-        /// <summary>
-        /// Initializes a new instance of the <see cref="ImageFrame{TPixel}" /> class.
-        /// </summary>
-        /// <param name="memoryManager">The <see cref="MemoryManager"/> to use for buffer allocations.</param>
-        /// <param name="size">The <see cref="Size"/> of the frame.</param>
-        /// <param name="metaData">The meta data.</param>
-        internal ImageFrame(MemoryManager memoryManager, Size size, ImageFrameMetaData metaData)
-            : this(memoryManager, size.Width, size.Height, metaData) {
+            if (!default(TPixel).Equals(backgroundColor))
```
Comment (antonfirsov) [inline src/ImageSharp/ImageFrame{TPixel}.cs]: Here it is I think!
You need to always clear the buffer.

Comment (JimBobSquarePants) [reply src/ImageSharp/ImageFrame{TPixel}.cs]: I'm just looking at that exact segment. Thought it might be that! 😄

**Pull request merged**